arch/src/x86_64/tdx/mod.rs

```diff
@@ -197,14 +197,14 @@ impl Default for PayloadImageType {
     }
 }
 
-#[repr(C)]
+#[repr(C, packed)]
 #[derive(Copy, Clone, Default, Debug)]
 pub struct PayloadInfo {
     pub image_type: PayloadImageType,
     pub entry_point: u64,
 }
 
-#[repr(C)]
+#[repr(C, packed)]
 #[derive(Copy, Clone, Default, Debug)]
 struct TdPayload {
     guid_type: HobGuidType,
```

Comment:
Do you also need to mark the type it embeds?

Reply:
Actually we don't need to make `TdPayload`  as packed